core/store/src/trie/mod.rs

```diff
@@ -1711,17 +1712,51 @@ impl Trie {
         &self,
         boundary_account: &AccountId,
         retain_mode: RetainMode,
-        opts: AccessOptions,
-    ) -> Result<StateRoot, StorageError> {
-        let mut trie_update = TrieStorageUpdate::new(&self);
-        let root_node = self.move_node_to_mutable(&mut trie_update, &self.root, opts)?;
-        trie_update.retain_split_shard(boundary_account, retain_mode, opts);
-        #[cfg(test)]
-        {
-            self.memory_usage_verify(&trie_update, GenericNodeOrIndex::Updated(root_node.0));
+    ) -> Result<TrieChanges, StorageError> {
+        if self.memtries.is_some() {
```

Comment:
mini nit: invert the logic and return early to flatten the code a bit

Reply:
maybe even with 
```
let Some(memtries) = self.memtries else {
  // no memtries handling
}
```

Reply:
Umm, we do have an else case, not sure if this comment holds?

Reply:
Yeah you can do else like I shared in the comment above. It's best for short else clauses here it's borderline. Overall I find it nicer to deal with the "strange" case first and then focus on the typical case outside of the if. It's also always good to unwrap Optional by matching rather than if and literal unwrap.

Reply:
Yes definitely makes sense, but I consider both the cases to be kinda equal. Anyway, what I did was to split the function into two parts for better readability, similar to how we have the `update` function above.